elements/dash-accordion/dash-accordion.js

```diff
@@ -0,0 +1,529 @@
+      this.addEventListener('accordion-last', this._onAccordionLast);
+
+      // Before the elements starts booting, it waits for
+      // the both `<dash-accordion-heading>` and `<dash-accordion-panel>` to
+      // load.
+      Promise.all([
+        customElements.whenDefined('dash-accordion-heading'),
+        customElements.whenDefined('dash-accordion-panel'),
+      ]).then(_ => {
+        // Acquire all headings and panels inside the element
+        const headings = this._allHeadings();
+        const panels = this._allPanels();
+        // If there are no headings, there is no way to activate panels.
+        // Abort.
+        if (headings.length === 0) return;
+
+        // Set up `aria-controls` and `aria-labelledby` on headings and panels.
+        headings.forEach(heading => {
+          const panel = heading.nextElementSibling;
+          if(!panel || panel.tagName !== 'DASH-ACCORDION-PANEL') {
+            console.error(`Tab #${heading.id} is not sibling` +
+              `of a <dash-accordion-panel>`);
+            return;
+          }
+
+          heading.setAttribute('aria-controls', panel.id);
+          panel.setAttribute('aria-labelledby', heading.id);
+        });
+
+        // Set all the panels to the collapsed state. This is done directly
+        // (and not using `_collapsePanel()`) to skip the animation.
+        panels.forEach(panel => {
+          panel.classList.remove('expanded');
+          panel.classList.add('collapsed');
+          panel.setAttribute('aria-hidden', 'true');
+        });
+        // Check if any of the headings have been marked as
+        // expanded. If so, all the associated panels get expanded.
+        headings
+          .filter(heading =>
+            heading.getAttribute('aria-expanded') === 'true')
+          .forEach(heading => {
+            const panel = this._panelForHeading(heading);
+            this._expandPanel(panel);
+          });
+      });
+    }
+
+    /**
+     * `disconnectedCallback` removes all the event listeners that
+     * `connectedCallback` added.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('accordion-expand', this._onAccordionExpand);
+      this.removeEventListener('accordion-collapse', this._onAccordionCollapse);
+      this.removeEventListener('accordion-previous', this._onAccordionPrevious);
+      this.removeEventListener('accordion-next', this._onAccordionNext);
+      this.removeEventListener('accordion-first', this._onAccordionFirst);
+      this.removeEventListener('accordion-last', this._onAccordionLast);
+    }
+
+    /**
+     * `_allPanels` returns all the panels in the accordion. This function could
+     * memoize the result if the DOM queries ever become a performance issue.
+     * The downside of memoization is that dynamically added headings and panels
+     * will not be handled.
+     *
+     * This is a method and not a getter, because a getter implies that it is
+     * cheap to read.
```

Comment:
I feel like someone seeing this might then wonder how expensive this method is. Not sure if there's a better way to phrase it though